gui/freehand.py

```diff
@@ -657,9 +657,8 @@ def _process_queued_event(self, tdw, event_data):
             else:
                 drawstate.avgtime = (tavg, nevents)
 
-        # Refuse drawing if the layer is locked or hidden
         current_layer = model._layers.current
-        if current_layer.locked or not current_layer.visible:
+        if not current_layer.get_paintable():
```

Comment:
It looks like this line is redundant, since it's not possible to queue an event when the current layer is not paintable.

Yes, currently, if you will use a brush with gigantic size, it's possible to change layers while FreehandMode process the queue. In this case, parts of a stroke will be on separate layers. But I think that such a scenario is uncommon, so it's not critical issue (for example, compared to the #699). Anyway, the `BrushworkModeMixin` is responsible for creating brush strokes, and it's always place them on the current layer. Fixing this requires quite big changes, which is out of scope of this PR — we need a separate ticket on the issue tracker.